lib/apiservers/engine/backends/container.go

```diff
@@ -770,13 +783,60 @@ func UnmapPorts(hostconfig *containertypes.HostConfig) error {
 			return err
 		}
 
+		// bridge-to-bridge pin hole for traffic from containers for exposed port
+		if err = interBridgeTraffic(portmap.Unmap, p.strHostPort, "", "", ""); err != nil {
+			return err
+		}
+
 		// update mapped ports
 		delete(containerByPort, p.strHostPort)
 		log.Debugf("unmapped port %s", p.strHostPort)
 	}
 	return nil
 }
 
+// interBridgeTraffic enables traffic for exposed port from one bridge network to another
+func interBridgeTraffic(op portmap.Operation, hostPort, proto, containerAddr, containerPort string) error {
+	switch op {
+	case portmap.Map:
+		switch proto {
+		case "udp", "tcp":
+		default:
+			return fmt.Errorf("unknown protocol: %s", proto)
+		}
+
+		// rule to allow connections from bridge interface for the
+		// specific mapped port. has to inserted at the top of the
+		// chain rather than appended to supersede bridge-to-bridge
+		// traffic blocking
+		args := []string{"-t", string(iptables.Filter),
+			"-i", bridgeIfaceName,
+			"-o", bridgeIfaceName,
+			"-p", proto,
+			"-d", containerAddr,
+			"--dport", containerPort,
+			"-j", "ACCEPT",
+		}
+
+		if _, err := iptables.Raw(append([]string{string(iptables.Insert), "VIC", "1"}, args...)...); err != nil && !os.IsExist(err) {
+			return err
+		}
+
+		btbRules[hostPort] = args
+	case portmap.Unmap:
+		if args, ok := btbRules[hostPort]; ok {
+			args = append([]string{string(iptables.Delete), "VIC"}, args...)
+			if _, err := iptables.Raw(args...); err != nil && !os.IsNotExist(err) {
+				return err
+			}
+
+			delete(btbRules, hostPort)
```

Comment:
`interBridgeTraffic` is only called with `cpbLock` being held in the calling function.